specs/themes/ThemeSettings.md

```diff
@@ -0,0 +1,157 @@
+
+  m_themeSettings.Changed([xamlRoot](const ThemeSettings& ts, const auto&) {
+    if (ts.HighContrast())
+    {
+      ::OutputDebugString(L"High contrast is ON, scheme is: ");
+      ::OutputDebugString(ts.HighContrastScheme().c_str());
+    }
+    else
+    {
+      ::OutputDebugString(L"High contrast is OFF.\n");
+    }
+  });
+}
+```
+
+When the app releases all its references to a **ThemeSettings** object, the object will be destroyed and the **Changed** event will
+no longer fire.
+
+### Methods
+|Name|Description|
+|-|-|
+|static CreateForWindowId(WindowId)|Creates a ThemeSettings object that will provide theme information for the given WindowId.|
+
+### Events
+|Name|Description|
+|-|-|
+|Changed|Fired when one of the object's properties has changed.|
+
+### Properties
+| Name | Description |
+|-|-|
+| Boolean HighContrast | True if High Contrast mode is enabled on the system. |
+| String HighContrastScheme | If HighContrast is true, a string from the system representing the High Contrast scheme, or style. |
+
+## ThemeSettings.CreateForWindowId static method
+
+``` c#
+static CreateForWindowId(WindowId windowId)
+```
+
+Creates a **ThemeSettings** object that will provide theme information for the given **WindowId**.
+
+The given **WindowId** must represent a window that:
+* is a top-level window,
+* AND is on the same process and thread as the caller.
+
+For example, you can get a **WindowId** from the **ContentEnvironment.AppWindowId** property.  You can get a **ContentEnvironment**
+object from a **XamlRoot** object.
+
+## ThemeSettings.Changed event
+
+```c#
+event TypedEventHandler<ThemeSettings, Object> Changed
+```
+
+Fired when one of the object's properties has changed.
+
+The **Changed** event will only fire while the **ThemeSettings** object is alive, so apps must ensure
+they hold a reference to the object as long as they want to receive the event.
+
+The **Changed** event will only fire as long as the window represented by **WindowId** is alive (has not yet received `WM_NCDESTROY`).
+
+# API Details
+
+_Spec note: this is a new namespace, **Microsoft.UI.System**.
+The predecessor for this type is `AccessibilitySettings`,
+which is in the `Windows.UI.ViewManagement` namespace,
+which is all UWP types we don't plan to carry into Desktop.
+This is a UI type that doesn't fit into any of the children of the `UI` namespace.
+We anticipate more types here, but don't have any current plans._
+
+
+``` c# (really midl3)
+namespace Microsoft.UI.System
+{
+  runtimeclass ThemeSettings 
+  {
+    static ThemeSettings CreateForWindowId(WindowId windowId);
+
+    event TypedEventHandler<ThemeSettings, Object> Changed;
+
+    Boolean HighContrast { get; }
+    String HighContrastScheme { get; } 
+  }
+}
+```
+
+# Appendix
+
+## Open Issues and Implementation Issues
+In the experimental API today, HighContrastScheme returns a stale value if HighContrast is off.  We should change this to return
+the empty string.
```

Comment:
Discussion: This is the behavior of the underlying system API and so we shouldn't try to policy override it here. Also, this API will be available on Win10 where the Settings UI gives this meaning (see earlier comment)